test/evp_kdf_test.c

```diff
@@ -1260,6 +1260,16 @@ static int test_kdf_sshkdf(void)
     return ret;
 }
 
+static int test_kdfs_same( EVP_KDF *kdf1, EVP_KDF *kdf2)
+{
+    /* Fast path in case the really are the same algorithm pointer */
+    if (kdf1 == kdf2)
+        return 1;
+    /* Compare their names and providers instead */
```

Comment:
Maybe explain why this is necessary - i.e When caching is disabled -

Reply:
Good point.